src/com/owncloud/android/ui/preview/PreviewTextFragment.java

```diff
@@ -0,0 +1,391 @@
+     */
+    public PreviewTextFragment() {
+        super();
+        mAccount = null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        super.onCreateView(inflater, container, savedInstanceState);
+        Log_OC.e(TAG, "onCreateView");
+
+
+        View ret = inflater.inflate(R.layout.text_file_preview, container, false);
+
+        mTextPreview = (TextView) ret.findViewById(R.id.text_preview);
+
+        return ret;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        OCFile file = getFile();
+
+        Bundle args = getArguments();
+
+        if (file == null)
+            file = args.getParcelable(FileDisplayActivity.EXTRA_FILE);
+
+        if (mAccount == null)
+            mAccount = args.getParcelable(FileDisplayActivity.EXTRA_ACCOUNT);
+
+
+        if (savedInstanceState == null) {
+            if (file == null) {
+                throw new IllegalStateException("Instanced with a NULL OCFile");
+            }
+            if (mAccount == null) {
+                throw new IllegalStateException("Instanced with a NULL ownCloud Account");
+            }
+        } else {
+            file = savedInstanceState.getParcelable(EXTRA_FILE);
+            mAccount = savedInstanceState.getParcelable(EXTRA_ACCOUNT);
+        }
+        setFile(file);
+        setHasOptionsMenu(true);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putParcelable(PreviewImageFragment.EXTRA_FILE, getFile());
+        outState.putParcelable(PreviewImageFragment.EXTRA_ACCOUNT, mAccount);
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        Log_OC.e(TAG, "onStart");
+    }
+
+    private void loadAndShowTextPreview() {
+        new TextLoadAsyncTask().execute(getFile().getStoragePath(), mTextPreview);
+    }
+
+    /**
+     * Reads the file to preview and shows its contents. Too critical to be anonymous.
+     */
+    private class TextLoadAsyncTask extends AsyncTask<Object, Void, StringWriter> {
+        private final String DIALOG_WAIT_TAG = "DIALOG_WAIT";
+        private TextView mTextView;
+
+        @Override
+        protected void onPreExecute() {
+            showLoadingDialog();
+        }
+
+        @Override
+        protected StringWriter doInBackground(java.lang.Object... params) {
+            if (params.length != 2)
+                throw new IllegalArgumentException("The parameters to " + TextLoadAsyncTask.class.getName() + " must be (1) the file location and (2) the text view to update");
+            final String location = (String) params[0];
+            mTextView = (TextView) params[1];
```

Comment:
Still would prefer to set the TextView reference in a constructor; the semantics of AsyncTask grant that fields set in constructor or ```onPreExecute``` are 'done' before ```doInBrackground``` is started.